src/object_ref.jl

```diff
@@ -1,12 +1,60 @@
 mutable struct ObjectRef
-    oid::ray_jll.ObjectIDAllocated
+    # oid::ray_jll.ObjectIDAllocated
+    oid_hex::String
```

Comment:
Can you elaborate on why we should store the hex string instead of the `oid`? maybe we should store both?

Reply:
yup.  was getting some wacky segfaults that went away when I did it this way.  I think probably they were due to double deallocation after copying.  So, we could keep the ObjectID around but we'd need to special case handling of that every time we create an instance.  Even on normal construction, if you construct >1 object ref from a single ObjectID instance, if you don't create a new instance then when you go to finalize the last one living you'll be accessing memory that's already been freed.

Basically it just feels safer to construct an ObjectID on demand when we need it, rather than trying to save a tiny amount of extra allocations by holding onto an instance that's managed by C++.